src/components/action-bar/action-bar.tsx

```diff
@@ -0,0 +1,237 @@
+        return (
+            <limel-action-bar-item
+                item={item}
+                onSelect={this.handleSelect}
+                isVisible={this.isVisible(index)}
+            />
+        );
+    };
+
+    private renderOverflowMenu = (items: Array<MenuItem | ListSeparator>) => {
+        if (!(this.actions.length - this.overflowCutoff)) {
+            return;
+        }
+
+        return (
+            <limel-action-bar-overflow-menu
+                openDirection={this.openDirection}
+                items={items}
+                onSelect={this.handleSelect}
+            />
+        );
+    };
+
+    private isVisible(index: number) {
+        return index < this.overflowCutoff;
+    }
+
+    private handleSelect = (
+        event: CustomEvent<ActionBarItem | ListSeparator>
+    ) => {
+        event.stopPropagation();
+        if (isItem(event.detail)) {
+            this.itemSelected.emit(event.detail);
+        }
+    };
+
+    private handleIntersection = (entries: IntersectionObserverEntry[]) => {
+        const intersectingItems = entries.filter(
+            (entry) => entry.isIntersecting
+        );
+
+        const notIntersectingItems = entries.filter(
+            (entry) => !entry.isIntersecting
+        );
+
+        if (this.firstRender) {
+            this.overflowCutoff = intersectingItems.length;
+        } else {
+            this.overflowCutoff =
+                this.overflowCutoff +
+                intersectingItems.length -
+                notIntersectingItems.length;
+        }
+
+        this.firstRender = false;
+    };
+
+    private createIntersectionObserver() {
+        const options = {
+            root: this.host.shadowRoot.querySelector('.items'),
+            rootMargin: '0px',
+            threshold: 1.0,
+        };
+
+        this.intersectionObserver = new IntersectionObserver(
+            this.handleIntersection,
+            options
+        );
+
+        this.host.shadowRoot
+            .querySelectorAll('limel-action-bar-item')
+            .forEach((actionBarItem) => {
+                this.observe(actionBarItem);
+            });
+    }
+
+    private observe(actionBarItem: HTMLLimelActionBarItemElement) {
+        this.intersectionObserver.observe(actionBarItem);
+        this.actionBarItems.push(actionBarItem);
+    }
+
+    private haveItemsChanged() {
+        this.actionBarItems.forEach(
+            (actionBarItem: HTMLLimelActionBarItemElement) => {
+                if (!this.host.shadowRoot.contains(actionBarItem)) {
+                    return true;
+                }
+            }
+        );
+
+        this.host.shadowRoot
+            .querySelectorAll('limel-action-bar-item')
+            .forEach((actionBarItem: HTMLLimelActionBarItemElement) => {
+                if (!this.actionBarItems.includes(actionBarItem)) {
+                    return true;
+                }
+            });
+
+        return false;
+    }
```

Comment:
This method will always return `false`. The inner returns from the `forEach` calls will never return from the `haveItemsChanged` method. We can replace the `forEach` calls with `find` or `some` which will return a value which we can check before returning from `haveItemsChanged`

Reply:
fixed in 5f115dc7e947662a23e081198e90f730534593fa